internal/server/server.go

```diff
@@ -27,7 +27,8 @@ type Request struct {
 	NonceHash         *NonceHash
 
 	// User Configuration
-	AuthHandler func(username string, realm string, srcAddr net.Addr) (key []byte, ok bool)
+	AuthHandler  func(username string, realm string, srcAddr net.Addr) (key []byte, ok bool)
+	QuotaHandler func(username string, realm string, srcAddr net.Addr) (ok bool)
```

Comment:
I think you should be returning `error` here.

Reply:
The reason why we decided for this API is (1) the error returned by the user is largely ignored by the server (we always [return](https://github.com/l7mp/turn/blob/28967fafea780757536c2a41463994d01924db2a/internal/server/turn.go#L160) the same standard STUN error code `stun.CodeAllocQuotaReached` to the client regardless of the error) and (2) because this is very similar to the `AuthHandler` API. Happy to change though.

Reply:
> I think you should be returning error here. There should be a specific error value (tested with error.Is) that causes the server to return Quota Exceeded, and the other values should cause it to return Internal server error (or the TURN equivalent, I haven't checked).

`AuthHandler` has the exact same structure in that it returns a bool instead of an error. There's an [in-progress PR](https://github.com/pion/turn/pull/415/files) to redo the design, should we change the signature there to likewise return an error?